src/rest-server/deploy/legacy_user_migrate.py

```diff
@@ -295,32 +368,39 @@ def main():
     transferCli = TransferClient(etcd_uri, args.k8sUri.replace('http://',''), admin_group_name, in_cluster)
 
     if transferCli.check_transfer_flag() is False and in_cluster is False:
-      etcd_user_list = transferCli.etcd_data_parse()
-      if etcd_user_list:
-          transferCli.prepare_secret_base_path()
-          for user in etcd_user_list:
-              secret_post_data = transferCli.secret_data_prepare(user)
-              transferCli.create_secret_user(secret_post_data)
-      else:
-          logger.info("No legacy data found")
-      http_post(transferCli.etcd_conn, transferCli.flag_path)
-      logger.info('Legacy user data transfer from etcd to kubernetes secret (pai-user namespace) successfully')
+        etcd_user_list = transferCli.etcd_data_parse()
+        if etcd_user_list:
+            transferCli.prepare_secret_base_path()
+            for user in etcd_user_list:
+                secret_post_data = transferCli.secret_data_prepare(user)
+                transferCli.create_secret_user(secret_post_data)
+        else:
+            logger.info("No legacy data found")
+        http_post(transferCli.etcd_conn, transferCli.flag_path)
+        logger.info('Legacy user data transfer from etcd to kubernetes secret (pai-user namespace) successfully')
     else:
-      logger.info("Etcd data has already been transferred to k8s secret")
+        logger.info("Etcd data has already been transferred to k8s secret")
 
     res = transferCli.prepare_secret_base_path_v2()
     if res[0] == 1 and res[1] == 1:
-      ns_pai_user_list = transferCli.namespace_v1_data_prepare()
-      for user in ns_pai_user_list:
-          secret_post_data = transferCli.secret_data_prepare_v2(user)
-          transferCli.create_secret_user_v2(secret_post_data)
-      vc_set = transferCli.vc_set
-      for vc in vc_set:
-          secret_post_data = transferCli.secret_data_prepare_v2_group(vc)
-          transferCli.create_secret_group_v2(secret_post_data)
-      logger.info('Legacy user data transfer from namespace v1 to namespace v2 successfully')
+        ns_pai_user_list = transferCli.namespace_v1_data_prepare()
+        for user in ns_pai_user_list:
+            secret_post_data = transferCli.secret_data_prepare_v2(user)
+            transferCli.create_secret_user_v2(secret_post_data)
+        vc_set = transferCli.vc_set
+        for vc in vc_set:
+            secret_post_data = transferCli.secret_data_prepare_v2_group(vc)
+            transferCli.create_secret_group_v2(secret_post_data)
+        logger.info('Legacy user data transfer from namespace v1 to namespace v2 successfully')
     else:
-      logger.info("Legacy data has already been transferred from v1 to v2. Skip it.")
+        logger.info("Legacy data has already been transferred from v1 to v2. Skip it.")
+
+    group_list, vc_set = transferCli.load_v2_groups()
```

Comment:
Will this logic happen everytime if we start rest-server? I have some concerns about the result if this script executed several times. Have you consider it?

Reply:
No, it will be checked and only be converted once, see line 164